File: Samples/VC/Samples/Sample1/Sample1.cpp
// Sample1.cpp : A simple C++ example using DataServer.dll COM object.
//
// Instantiates DataServer, loads a configuration file, and acquires data.

#include "stdafx.h"
#include <objbase.h>

// import data server 
// TODO: change path to DataServer.dll
#import "C:\Projects\Biomechanics\DataServer\DataServer\DataServer\Release\DataServer.dll"

int _tmain(int argc, _TCHAR* argv[])
{
	CoInitialize (NULL);
	
	// Build Scope for COM CoInitialize/CoUnitialize
	{ 
		// local variable to track completion.
		ULONG prevscan (0);
	
		// Create the server object
		DataServerLib::IDataServerPtr server;
		server.CreateInstance( L"DataServer.DataServer" );
		if( server==NULL )
			return -1;
		
		try
		{
			// Get DAQ object, set Configuration file, board type, and Instacal Board Number
			// TODO: 
			// - change path to config.xml if needed. (parameter 1)
			// - set board type (parameter 2)
			// - set board number from InstaCal (parameter 3)
			DataServerLib::IDaqControlPtr daq;
			daq = server->CreateDaqObject (L"../../../XML/config.xml", DataServerLib::brdKistler5691A1, 0);

			// set amplifier to on 
			daq->MeasureOn ();

			// read offets if enabled
			daq->ReadOffsets (DataServerLib::rngBIP10VOLTS);
						
			// prepare data acqusition, and wait for software trigger event
			// Start (long rate_per_channel, long samples_per_channel, enum trigger_option, enum daq_range)
			daq->Start (1000, 5000, DataServerLib::trigImmediate, DataServerLib::rngBIP10VOLTS); 

			// loop while running
			while( daq->Running ) {
				// yield to other threads, let them do some work
				Sleep (50);
					
				// check current amount available
				ULONG thisscan (daq->LastAvailableScan); 

				if( thisscan>prevscan ) {
					// call to process the data
					// ProcessData (prevscan, thisscan);
					prevscan = thisscan;
					wprintf (L"processed %d...\n", thisscan);
				}
			}
					
			// call stop (although, it should already be stopped)
			daq->Stop ();
					
			// place plate into reset
			daq->MeasureOff ();
		}
		catch( _com_error e ) 
		{ 
			wprintf ((LPCTSTR)e.Description());
		}
	}

	CoUninitialize();
}

File: Samples/VB/Samples/Sample1/Sample1.vb
Imports DataServerLib

Public Class Sample1
    ' Declare a member variable for the Daq control interface
    Private daq As DaqControl

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        ' declare a dataserver object, get a DaqControl object from server factory
        Dim server As New DataServer

        ' create object, assigning a configuration file, board type, and Instacal board number
        ' TODO:
        ' set path to XML configuration file
        ' set board type and MCC Instacal board number
        daq = server.CreateDaqObject("..\..\..\..\..\XML\config.xml", BoardType.brdKistler5691A1, 0)

        ' enable measure
        daq.MeasureOn()

        ' read offsets
        daq.ReadOffsets(RangeType.rngBIP10VOLTS)
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        ' set to reset
        daq.MeasureOff()
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        ' Start acquisition
        ' 1000Hz for 5 seconds
        daq.Start(1000, 5000, TriggerType.trigImmediate, RangeType.rngBIP10VOLTS)

        ' Wait until done
        Do Until Not daq.Running()
        Loop
    End Sub
End Class
